Source/Core/Core/FifoPlayer/FifoPlayer.cpp

```diff
@@ -103,6 +103,9 @@ bool FifoPlayer::Play()
 				++m_CurrentFrame;
 			}
 		}
+
+		// Let the waiting thread know we are done leaving
+		PowerPC::FinishStateMove();
```

Comment:
I doubt this should be placed within the while (PowerPC::GetState() != PowerPC::CPU_POWERDOWN) loop. May you try to move it to the next line?